File: src/main/java/de/grimsi/gameyfin/views/MainView.java
package de.grimsi.gameyfin.views;

import com.vaadin.flow.component.html.H1;
import com.vaadin.flow.component.html.Pre;
import com.vaadin.flow.component.orderedlayout.VerticalLayout;
import com.vaadin.flow.router.PageTitle;
import com.vaadin.flow.router.Route;
import de.grimsi.gameyfin.layouts.MainLayout;
import jakarta.annotation.security.PermitAll;


@Route(value = "", layout = MainLayout.class)
@PageTitle("Gameyfin")
@PermitAll
public class MainView extends VerticalLayout {

    public MainView() {
        add(new H1("Gameyfin main page"));
        add(new Pre("Work in progress"));
    }
}

File: src/main/kotlin/de/grimsi/gameyfin/ui/views/MainView.kt
package de.grimsi.gameyfin.ui.views

import com.github.mvysny.karibudsl.v10.h1
import com.github.mvysny.karibudsl.v10.pre
import com.github.mvysny.karibudsl.v10.verticalLayout
import com.vaadin.flow.component.orderedlayout.VerticalLayout
import com.vaadin.flow.router.Route
import de.grimsi.gameyfin.ui.layouts.MainLayout
import jakarta.annotation.security.PermitAll


@Route("", layout = MainLayout::class)
@PermitAll
class MainView : VerticalLayout() {

    init {
        verticalLayout {
            h1 { text = "Gameyfin main page" }
            pre { text = "Work in progress" }
        }
    }
}